votrfront/js/MojeSkuskyPage.js

```diff
@@ -94,11 +94,147 @@ export var MojeSkuskyPageContent = React.createClass({
+      // verificator: http://severinghaus.org/projects/icv/
+      var header = [
+        "BEGIN:VCALENDAR",
+        "VERSION:2.0",
+        "PRODID:-//svt.fmph.uniba.sk//NONSMGL  votr-170615//",
+        "X-WR-CALNAME:Moje termíny hodnotenia",
+        "X-WR-CALDESC:Kalendár skúšok vyexportovaný z aplikácie VOTR",
+        "X-WR-TIMEZONE:Europe/Bratislava",
+      ].join("\r\n");
+
+      var events = {};
+      
+      function createVevent(termin) {
+        var header = "BEGIN:VEVENT";
+
+        // unique identificator for each event (so we can identify copies of the same event)
+        var uid = termin.termin_key + "@" + "votr.uniba.sk";
+
+        // DTSTAMP is something like LAST-MODFIFIED column (mandatory), must be in UTC time format
+        function getUTCTimeFormat(date) {
+          // dirty hack, as I don't now any normal way to convert Date() to UTC time format: YYYYMMDDTHHMMSSZ (Zulu time)
+          var s = date.toISOString();
+          var re = /(\d*)-(\d*)-(\d*)T(\d*):(\d*):(\d*)(\.\d*)*Z/;
+          var ss = s.replace(re, "$1$2$3T$4$5$6Z");
+          return ss;
+        }
+        var dtstamp = getUTCTimeFormat(new Date());
+
+        var [den, mesiac, rok] = termin.datum.split(".");
+        var [hodina, minuty] = termin.cas.split(":");
+        var dtstart = `${rok}${mesiac}${den}T${hodina}${minuty}00`;
+
+        // as for there is no info about duration, we'll set it for 4 hours
+        var hodina_koniec = (parseInt(hodina) + 4).toString();
+        // add leading zero
+        if (hodina_koniec.length == 1) {
+          hodina_koniec = "0" + hodina_koniec;
+        }
+        var dtend = `${rok}${mesiac}${den}T${hodina_koniec}${minuty}00`;
+
+        function escapeSymbols(s) {
+          return s.replace(/\\n/g, "\\n");
+        }
+
+        //@TODO ake vsetky informacie chceme zobrazovat v popise eventu? (zatial su take, ako vo FAJR)
+        var desc = [
+          "Prihlasovanie:" + escapeSymbols(termin.prihlasovanie), 
+          "Odhlasovanie:" + escapeSymbols(termin.odhlasovanie),
+          "Poznámka:" + escapeSymbols(termin.poznamka),
+        ].join("\\n"); // double backquote is intentional
+
+        var fields = [
+          "SUMMARY:" + termin.nazov_predmetu,
+          "UID:" + uid,
+          "DTSTAMP:" + dtstamp,
+          "DTSTART;TZID=Europe/Bratislava:" + dtstart,
+          "DTEND;TZID=Europe/Bratislava:" + dtend,
+          "DESCRIPTION:" + desc,
+        ];
+
+        if (termin.miestnost !== null && termin.miestnost !== undefined && termin.miestnost.length > 0) {
+          fields.push("LOCATION:" + termin.miestnost);
+        }
+
+        var footer = "END:VEVENT";
+        return [header, fields.join("\r\n"), footer].join("\r\n");
+      }
+
+      terminy.forEach((termin) => events[termin.termin_key] = createVevent(termin));
+      events = _.values(events);
+
+      var footer = "END:VCALENDAR";
+      return [header, events.join("\r\n"), footer].join("\r\n");
```

Comment:
Blé, toľko vnorených joinov, prečo nie proste pole "riadky" a na konci `return riadky.join("\r\n")`?